src/lib/select/select.ts

```diff
@@ -725,15 +731,18 @@ export class MdSelect implements AfterContentInit, OnDestroy, OnInit, ControlVal
 
   /** Calculates the scroll position and x- and y-offsets of the overlay panel. */
   private _calculateOverlayPosition(): void {
-    const panelHeight =
-        Math.min(this.options.length * SELECT_OPTION_HEIGHT, SELECT_PANEL_MAX_HEIGHT);
-    const scrollContainerHeight = this.options.length * SELECT_OPTION_HEIGHT;
+    const items = this._getItemCount();
+    const panelHeight = Math.min(items * SELECT_ITEM_HEIGHT, SELECT_PANEL_MAX_HEIGHT);
+    const scrollContainerHeight = items * SELECT_ITEM_HEIGHT;
 
     // The farthest the panel can be scrolled before it hits the bottom
     const maxScroll = scrollContainerHeight - panelHeight;
 
     if (this._selectionModel.hasValue()) {
-      const selectedIndex = this._getOptionIndex(this._selectionModel.selected[0]);
+      let selectedIndex = this._getOptionIndex(this._selectionModel.selected[0]);
+
+      selectedIndex += this._getLabelCountBeforeOption(selectedIndex);
```

Comment:
Nit: Since this isn't strictly the selectedIndex anymore, can we rename for clarity? `selectedOffset` or something?